contracts/ExitHandler.sol

```diff
@@ -87,21 +91,30 @@ contract ExitHandler is DepositHandler {
     require(exits[utxoId].amount == 0, "The exit for UTXO has already been started");
     require(!exits[utxoId].finalized, "The exit for UTXO has already been finalized");
 
-    // check youngest input tx inclusion in the root chain block
-    bytes32 inputTxHash;
-    uint64 inputTxPos;
-    (inputTxPos, inputTxHash,) = TxLib.validateProof(96, _youngestInputProof);
-    require(
-      inputTxHash == exitingTx.ins[_inputIndex].outpoint.hash, 
-      "Input from the proof is not referenced in exiting tx"
-    );
-    
     uint256 priority;
-    if (isNft(out.color)) {
-      priority = (nftExitCounter << 128) | uint128(utxoId);
-      nftExitCounter++;
-    } else {      
-      priority = getERC20ExitPriority(timestamp, utxoId, inputTxPos);
+    if (_youngestInputProof.length > 0) {
+      // check youngest input tx inclusion in the root chain block
+      bytes32 inputTxHash;
+      (txPos, inputTxHash,) = TxLib.validateProof(96, _youngestInputProof);
+      require(
+        inputTxHash == exitingTx.ins[_inputIndex].outpoint.hash, 
+        "Input from the proof is not referenced in exiting tx"
+      );
+      
+      if (isNft(out.color)) {
+        priority = (nftExitCounter << 128) | uint128(utxoId);
+        nftExitCounter++;
+      } else {      
+        priority = getERC20ExitPriority(timestamp, utxoId, txPos);
+      }
+    } else {
+      require(exitingTx.txType == TxLib.TxType.Deposit, "Expected deposit tx");
+      if (isNft(out.color)) {
+        priority = (nftExitCounter << 128) | uint128(utxoId);
+        nftExitCounter++;
+      } else {      
+        priority = getERC20ExitPriority(timestamp, utxoId, txPos);
```

Comment:
Timestamp is 0 here, which means that the deposit will always have the highest priority. So operator can include a deposit out of nowhere and it will have higher priority than valid utxos before it. Or am I missing something?

Reply:
good catch, fixed.